lib/from_openstudio/geometry/room.rb

```diff
@@ -96,6 +96,13 @@ def self.outdoor_shades_from_space(space)
         site_transformation = shading_surface_group.siteTransformation
         shading_surface_group.shadingSurfaces.each do |shading_surface|
           result << Shade.from_shading_surface(shading_surface, site_transformation)
+          # get shade construction if specified
+          shade_const_base = shading_surface.construction
+          unless shade_const_base.empty?
+            shade_const_obj = shade_const_base.get
+            shade_const = shade_const_obj.to_LayeredConstruction.get
+            $shade_construction << shade_const
```

Comment:
I feel like `$shade_construction` should be a hash where you organize the shade constructions by their unique name or ID and not a list with potentially a bunch of duplicated construction objects.  It's just going to make the run time a lot longer if you have all of these duplicated references to constructions in the list instead of one reference per unique construction. Also, is there a reason why you are doing this same copy/pasted check for each parent object (Room, Face, Aperture) instead of just having one check in the Shade class?

Reply:
Thanks Chris, that is a good suggestion.